Add support for environment variables in command

npostulart opened the pull request:
According to feature request https://github.com/stepci/stepci/issues/9 this PR adds support for environment variables set as argument. The arguments will override the settings in the workflow file.

mishushakov reviewed (changes requested):
Looks awesome! Some changes I'd suggest doing

1) On line 24, change env type to `EnvironmentVariables`

2) Change `envOverride` to `env`

3) Instead of overriding env variables defined in `workflow.env`, we should *merge* them instead. This is useful if I want to have pre-defined env variables, like hostname in my file, but want to supply additional env variables later on

Example:

```js
const env = { ...workflow.env, ...options?.env }
```

Also, the usage of `??` is smart! I didn't know such operator existed in JS. My life is different now 😂

npostulart commented:
> 1. On the line 24, change the type of `env` to `EnvironmentVariables`

Check, changed it.

> 2. Rename `envOverride` to `env`

Had it first as env and decided to change it to envOverride to make clear that it will override env variables in workflow. But I think both is fine, so I changed it back to env.

> 3. Instead of overriding env variables defined in `workflow.env`, we should **combine** them instead. This is useful if I want to have pre-defined env variables, like hostname in the file, but want to supply additional env variables later on
>    Example:
>    ```js
>    const env = { ...workflow.env, ...options?.env }
>    ```

I did not override the env variables completely but merge them as you already quote in your code example. That's what https://github.com/stepci/runner/pull/56/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R331 is for.

mishushakov reviewed (approved):
Looks cool to me 😎
We're accepting it! Thank you for your pull request!
